apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -378,3 +384,43 @@ func Labels(labels map[string]string) omfunc {
 		o.SetLabels(labels)
 	}
 }
+
+func GetAppInSameEnv(client client.Client, ctx context.Context, app *ClowdApp, appList *ClowdAppList) error {
+	err := client.List(ctx, appList)
```

Comment:
could we use a `client.MatchingFields` here to remove the need for the `for` loop below?

https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client#MatchingFields

Reply:
I fixed it for one, unfortunately, with the cache, you can only match on a single field, and I had to add a custom index for it as it is :)